geonode/harvesting/utils.py

```diff
@@ -53,3 +52,25 @@ def validate_worker_configuration(harvester_type, configuration: typing.Dict):
             jsonschema.validate(configuration, schema)
         except jsonschema.exceptions.SchemaError as exc:
             raise RuntimeError(f"Invalid schema: {exc}")
+
+
+def get_xpath_value(
+        element: etree.Element,
+        xpath_expression: str,
+        nsmap: typing.Optional[dict] = None
+) -> typing.Optional[str]:
+    if not nsmap:
+        nsmap = element.nsmap
+    values = element.xpath(f"{xpath_expression}//text()", namespaces=nsmap)
+    return "".join(values).strip() or None
+
+
+def get_xpath_elements(
+        element: etree.Element,
+        xpath_expression: str,
+        nsmap: typing.Optional[dict] = None
+) -> typing.List[etree.Element]:
+    if not nsmap:
+        nsmap = element.nsmap
+    elements = element.xpath(f"{xpath_expression}", namespaces=nsmap)
+    return elements
```

Comment:
If you can always get `nsmap` from the provided `element`, why have it also as an optional parameter? Is there an actual use case for this in the code? If not, then please make the function accept only two parameters - that will make it simpler (removing the `if` block inside) and easier to read.

Also, as stated above, it seems like the `get_xpath_elements()` function may not be necessary in this case. If you are only using it as an intermediary in order to then call `get_xpath_value()` on its result, then there is no need for that, as XPath allows nested queries. Otherwise, if you have a valid use case for it I won't mind having it.

I'm just trying to see if you can remove code instead of adding more - the more code we have, the harder it is for people to maintain it in the future.